internal/internal_event_handlers.go

```diff
@@ -1217,6 +1250,7 @@ func (weh *workflowExecutionEventHandlerImpl) handleWorkflowExecutionStarted(
 	if err != nil {
 		return err
 	}
+	weh.sdkFlags.tryUse(SDKFlagProtocolMessageCommand, !weh.isReplay)
```

Comment:
nit: Since you spent a lot of time debugging this so it might be worth documenting why we set this on start rather then first use

Reply:
(added in handleWorkflowExecutionStarted)